main/views.py

```diff
@@ -19,3 +27,49 @@ def form_valid(self, form):
         data['id_list'] = new_id_list
         form.data = data
         return self.render_to_response(self.get_context_data(form=form))
+
+
+def upload_file(request):
+    if request.method == 'POST':
```

Comment:
Why not use class-based views as with IndexView to avoid mess with `if request.method == ...`?